sdk/dapp-kit/src/hooks/wallet/useConnectWallet.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { UseMutationOptions } from '@tanstack/react-query';
+import { useMutation } from '@tanstack/react-query';
+import type {
+	StandardConnectInput,
+	StandardConnectOutput,
+	WalletAccount,
+} from '@mysten/wallet-standard';
+import { useWalletContext } from '../../components/WalletProvider.js';
+import { WalletAlreadyConnectedError, WalletNotFoundError } from '../../errors/walletErrors.js';
+import {
+	getMostRecentWalletConnectionInfo,
+	setMostRecentWalletConnectionInfo,
+} from '../../utils/walletUtils.js';
+import { walletMutationKeys } from '../../constants/walletMutationKeys.js';
+
+type ConnectWalletArgs = {
+	/** The name of the wallet as defined by the wallet standard to connect to. */
+	walletName: string;
+} & StandardConnectInput;
+
+type ConnectWalletResult = StandardConnectOutput;
+
+type UseConnectWalletMutationOptions = Omit<
+	UseMutationOptions<ConnectWalletResult, Error, ConnectWalletArgs, unknown>,
+	'mutationFn'
+>;
+
+/**
+ * Mutation hook for establishing a connection to a specific wallet.
+ */
+export function useConnectWallet({
+	mutationKey,
```

Comment:
pls make optional

Reply:
yep this is optional (see https://github.com/MystenLabs/sui/pull/13706/files#r1320242566)